components/Article.jsx

```diff
@@ -1,6 +1,38 @@
-import React from 'react'
+import React, { useState } from 'react'
+import {updateArticleVotes} from '../src/api'
+import { Navigate } from 'react-router-dom'
 
 const SingleArticle = ({ singleArticle }) => {
+  const [articleVotes, setArticlevotes] = useState(singleArticle.votes)
+  const [hasLiked, setHasLiked] = useState(false)
+  const [hasDisliked, setHasDisliked] = useState(false)
+  const [error, setError] = useState(false)
+
+  const handleVotes = (e) =>  {
+    if (e.target.innerText === 'Like' && !hasLiked) {
+      setArticlevotes((articleVotes) => articleVotes + 1)
+      setHasLiked(true)
+      setHasDisliked(false)
+      updateArticleVotes(singleArticle.article_id, 1)
+      .catch((err) => {
+        setError(true)
+      })
+    }
+    else if (e.target.innerText === 'Dislike' && !hasDisliked) {
+      setArticlevotes((articleVotes) => articleVotes - 1)
+      setHasDisliked(true)
+      setHasLiked(false)
+      updateArticleVotes(singleArticle.article_id, -1)
+      .catch((err) => {
+        setError(true)
```

Comment:
The logic above is 'sort of working'
This component will allow us to either vote up one and remove the vote - or - vote down one and remove that vote. It would be more conventional if we could do both: up vote and then change our mind and down vote.